packages/sftp-client/index.js

```diff
@@ -136,44 +121,15 @@ class SftpClient {
    * @returns {Promise<Array<Object>>} list of file objects
    */
   async list(remotePath) {
-    await this.connect();
-
-    return new Promise((resolve, reject) => {
-      this.sftp.readdir(remotePath, (err, list) => {
-        if (err) {
-          if (err.message.includes('No such file')) {
-            return resolve([]);
-          }
-          return reject(err);
-        }
-        return resolve(list.map((i) => ({
-          name: i.filename,
-          path: remotePath,
-          type: i.longname.substr(0, 1),
-          size: i.attrs.size,
-          time: i.attrs.mtime * 1000
-        })));
-      });
-      this.client.on('error', reject);
-    });
-  }
-
-  /**
-   * get readable stream of the remote file
-   *
-   * @param {string} remotePath - the full path to the remote file to be fetched
-   * @returns {Promise} readable stream of the remote file
-   * @private
-   */
-  async getReadableStream(remotePath) {
-    await this.connect();
-
-    return new Promise((resolve, reject) => {
-      const readStream = this.sftp.createReadStream(remotePath);
-      readStream.on('error', reject);
-      this.client.on('error', reject);
-      return resolve(readStream);
-    });
+    const remoteFiles = await this.sftp.list(remotePath);
+
+    return remoteFiles.map((remoteFile) => ({
```

Comment:
I'm just assuming that the properties returned here are equivalent to how they were being determined before?

Reply:
Correct.